src/Utils/util.test.ts

```diff
@@ -20,4 +21,201 @@ describe('util', () => {
+                createdDateTime: "2019-02-25T00:00:00.000Z",
+                isTerminal: true,
+                negativeConditions: [],
+                negativeEntities: [],
+                packageCreationId: 0,
+                packageDeletionId: 0,
+                payload: '{"json":{"kind":"value","document":{"kind":"document","data":{},"nodes":[{"kind":"block","type":"line","isVoid":false,"data":{},"nodes":[{"kind":"text","leaves":[{"kind":"leaf","text":"hi","marks":[]}]}]}]}}}"',
+                requiredConditions: [],
+                requiredEntities: [],
+                requiredEntitiesFromPayload: [],
+                suggestedEntity: null,
+                version: 0
+            }
+
+            let actionSet: CLM.ActionBase[] = []
+
+            // as first action created
+            expect(isNewActionUnique(sampleAction, actionSet)).toEqual(true)
+
+            // as absolutely identical action
+            actionSet.push(sampleAction)
+            expect(isNewActionUnique(sampleAction, actionSet)).not.toBe(true)
+
+            // as a similar, but not identical into very similar set
+            actionSet = [{ ...sampleAction }]
+            let similar = { ...sampleAction, isTerminal: !sampleAction.isTerminal }
+            expect(isNewActionUnique(similar, actionSet)).not.toBe(false)
+
+            // as a similar, but not identical into very similar set. how smart is the find operator?
+            actionSet = [sampleAction]
+            similar = { ...sampleAction, requiredEntities: ["85eccbec-7d01-4aea-a704-b2fdab09cf32"] }
+            expect(isNewActionUnique(similar, actionSet)).not.toBe(false)
+        })
+    })
+    describe('isNewActionUniqueENDSESSION', () => {
+        it('exercise ENDSESSION action creation', () => {
+            const sampleAction = {
+                actionId: "c64988df-a707-46c9-873c-8de42b9a116d",
+                actionType: CLM.ActionTypes.END_SESSION,
+                createdDateTime: "2019-02-25T00:00:00.001Z",
+                isTerminal: true,
+                negativeConditions: [],
+                negativeEntities: [],
+                packageCreationId: 0,
+                packageDeletionId: 0,
+                payload: '"{"json":{"kind":"value","document":{"kind":"document","data":{},"nodes":[{"kind":"block","type":"line","isVoid":false,"data":{},"nodes":[{"kind":"text","leaves":[{"kind":"leaf","text":"1","marks":[]}]}]}]}}}"',
+                requiredConditions: [],
+                requiredEntities: [],
+                requiredEntitiesFromPayload: [],
+                suggestedEntity: null,
+                version: 0
+            }
+
+            let actionSet: CLM.ActionBase[] = []
+
+            // as first action created
+            expect(isNewActionUnique(sampleAction, actionSet)).toEqual(true)
+
+            // as absolutely identical action
+            actionSet.push(sampleAction)
+            expect(isNewActionUnique(sampleAction, actionSet)).not.toBe(true)
```

Comment:
Use positive assertions when possible.  We want to know exactly what it should be not what it isn't.  It just so happens that it's a boolean and we know it can only be false but don't rely on this.

Reply:
Switching to positive asserts